### PR TITLE
apbs: init at 3.4.1

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes
Software for biomolecular electrostatics and solvation calculations.

homepage: https://www.poissonboltzmann.org/
source: https://github.com/Electrostatics/apbs

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/chemistry/apbs/default.nix [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    fetk
+    suitesparse
+    blas
+    python3
+  ] ++ lib.optionals (stdenv.isDarwin) [
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/apbs/default.nix:78]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```

File: pkgs/applications/science/chemistry/apbs/default.nix [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    owner = "Electrostatics";
+    repo = "apbs";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-2DnHU9hMDl4OJBaTtcRiB+6R7gAeFcuOUy7aI63A3gQ=";
+  };
+
+  postPatch = ''
+    # ImportFETK.cmake downloads source and builds fetk
+    substituteInPlace CMakeLists.txt --replace "include(ImportFETK)" ""
+    substituteInPlace CMakeLists.txt --replace 'import_fetk(''${FETK_VERSION})' ""
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/apbs/default.nix:66]:
```suggestion
    substituteInPlace CMakeLists.txt \
      --replace "include(ImportFETK)" "" \
      --replace 'import_fetk(''${FETK_VERSION})' ""
```

Comment (natsukium) [reply pkgs/applications/science/chemistry/apbs/default.nix]: I did not know that I could combine them into one.
Thank you.

File: pkgs/applications/science/chemistry/apbs/default.nix [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    owner = "Electrostatics";
+    repo = "apbs";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-2DnHU9hMDl4OJBaTtcRiB+6R7gAeFcuOUy7aI63A3gQ=";
+  };
+
+  postPatch = ''
+    # ImportFETK.cmake downloads source and builds fetk
+    substituteInPlace CMakeLists.txt \
+      --replace "include(ImportFETK)" ""
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/apbs/default.nix:66]:
```suggestion
      --replace "include(ImportFETK)" "" \
```

**Pull request merged**